File: Sources/structlyC/03_beginner_recursion/reverse_string_recursive.cpp
//
//  reverse_string_recursive.h
//  MyCLI
//
//  Created by new on 3/14/26.
//

#include <string>

std::string reverseString(std::string s) {
    if (s.size() == 0) {
        return "";
    }
    return reverseString(std::string(s.begin()+1, s.end())) + s[0];
}

File: Sources/structly/03_beginner_recursion/ReverseStringRecursive.swift
//
//  ReverseStringRecursive.swift
//  MyCLI
//
//  Created by new on 3/14/26.
//

func reverseString(_ s: String) -> String {
    func rev(sub: Substring) -> String {
        if sub.count <= 1 {
            return String(sub)
        }
        let firstPart = sub.dropFirst()
        let secondPart = String(sub.first!)
        return rev(sub: firstPart) + secondPart
    }
    return rev(sub: s[...])
}
